zerver/tests/test_auth_backends.py

```diff
@@ -658,6 +659,7 @@ def setUp(self) -> None:
         self.backend = self.BACKEND_CLASS
         self.backend.strategy = DjangoStrategy(storage=BaseDjangoStorage())
         self.user_profile.backend = self.backend
+        self.logger = logging.getLogger(f'zulip.auth.{self.backend.name}')
```

Comment:
how does this work? When we mock this `self.logger`, does that result in the loggers inside the actual backend instances to be mocked too? That's kind of neat if it works this, but I want to check if I understand correctly

Reply:
Yes, mocking that results in the actual backend instances to be also be mocked. This verification I did might make this clear. :slightly_smiling_face: 
```
>>> import logging
>>> logger1 = logging.getLogger('zulip.auth.asdf')
>>> logger2 = logging.getLogger('zulip.auth.asdf')
>>> logger1 == logger2
True
>>> def test():
...     logger1.info("logged")
... 
>>> import mock
>>> with mock.patch.object(logger2, 'info') as m:
...     test()
...     m.assert_called_once_with("logged")
... 
>>> with mock.patch.object(logger2, 'info') as m:
...     test()
...     m.assert_called_once_with("not logged")
... 
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/mock/mock.py", line 957, in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
  File "/srv/zulip-py3-venv/lib/python3.6/site-packages/mock/mock.py", line 944, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: expected call not found.
Expected: info('not logged')
Actual: info('logged')
```